packages/solarwinds-apm/src/hooks.es.js

```diff
@@ -14,13 +14,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { init } from "./init.js"
+import { isMainThread } from "node:worker_threads"
 
 // init in here too so that everything can be done through a single --loader flag
-try {
-  await init()
-} catch (err) {
-  console.warn(err)
+if (isMainThread) {
+  try {
+    const { init } = await import("./init.js")
+    await init()
+  } catch (err) {
+    console.warn(err)
+  }
 }
```

Comment:
In Node >20 the import hooks are registered in a worker thread instead of the main one. We want to initialise in the main thread, not a worker thread.